rdkit/Chem/PandasTools.py

```diff
@@ -432,14 +239,140 @@ def RenderImagesInAllDataFrames(images=True):
     to change the rendering only for a single dataframe use the "ChangeMoleculeRendering" method
     instead.
     '''
-  if images:
-    pd.core.frame.DataFrame.to_html = patchPandasHTMLrepr
-    if defPandasRepr is not None:
-      pd.core.frame.DataFrame._repr_html_ = patchPandasrepr
-  else:
-    pd.core.frame.DataFrame.to_html = defPandasRendering
-    if defPandasRepr is not None:
-      pd.core.frame.DataFrame._repr_html_ = defPandasRepr
+  try:
+    PandasPatcher.renderImagesInAllDataFrames(images)
+  except NameError:
+    log.warning("Failed to patch pandas - unable to change molecule rendering")
+
+
+try:
```

Comment:
Why not just try the `import pandas as pd` at the very top of this file and let the `ImportError` come up to the user?
There's no point importing this if you don't have pandas installed.

Reply:
I would change the logic that I have in place at the moment since it is quite complex and tested to work across a range of `pandas` version. My reasoning was mostly to enable access to `PrintAsImageString()` (formerly `PrintAsBase64PNGString()`) which has been part of the `PandasTools` module for a long time and someone might be using even without importing `pandas` (unlikely, but not impossible).

Reply:
We don't need to do it here, but I think it's worth making the change later.